pkg/install/install.go

```diff
@@ -26,10 +27,18 @@ type TemplateParameters struct {
 	Namespace          string
 	ManifestGeneration bool
 	AdditionalFluxArgs []string
+	ConfigFileContent  string
```

Comment:
Given that this package is users externally. The value expected for this field is really awkward (particularly given its name) since it must be YAML with a the name of the file as a key and the body of the file (base64 encoded) as a value.

If we only support a config file format then there is no need to specify the filename (we can hardcode it), then we can decide to encode it as base64 or not based on whether a secret is used.

Reply:
Yeah it's a bit of a burr on the install API that you have to know how to create the file content. It's a bit of a faff to get the templates to e.g., indent things reliably, but :crossed_fingers: possible.